lib/hasher/src/ADLER32Strategy.cpp

```diff
@@ -0,0 +1,79 @@
+#include "ADLER32Strategy.hpp"
+#include "checksum.hpp"
+
+#include <string>
+#include <sstream>
+#include <iostream>
+#include <iomanip>
+#include <string.h>
+#include <boost/algorithm/string/predicate.hpp>
+//#include <zlib.h>
+
+#include "base64.h"
+
+namespace irods {
+
+    struct adler32_parts {
+        uint32_t a;
+        uint32_t b;
+    };
+
+    adler32_parts adler32_init() {
+        return adler32_parts{1, 0};
+    }
+
+    static adler32_parts adler32_update(const adler32_parts& parts, const unsigned char *data, size_t len) {
+
+        const uint32_t MOD_ADLER = 65521;
+
+        uint32_t a = parts.a, b = parts.b;
+
+		// Process each byte of the data in order
+		for (size_t index = 0; index < len; ++index)
+		{
+			a = (a + data[index]) % MOD_ADLER;
+			b = (b + a) % MOD_ADLER;
+		}
+
+        return adler32_parts{a, b};
```

Comment:
indentation

Reply:
I had accidentally copied in some tab chars which looked fine with my tabstop setting.  Replaced with four spaces.